packages/core/src/state-management/repository.ts

```diff
@@ -50,12 +50,18 @@ export class Repository {
    */
   stateManager: StateManager;
 
-  constructor(limit: number, private readonly logger: DiagnosticsLogger) {
+  /**
+   * @param cacheLimit - Maximum number of documents to store in memory cache.
+   * @param logger - Where we put diagnostics messages.
+   * @param concurrencyLimit - Maximum number of concurrently running tasks on the documents.
+   */
+  constructor(cacheLimit: number, private readonly logger: DiagnosticsLogger, concurrencyLimit?: number) {
```

Comment:
Any reason for `concurrencyLimit` to be optional?  I generally prefer to avoid optional parameters in internal-only APIs so that behavior is clearer at the call site and developers have to consider all available options whenever calling something

Reply:
Yup. Concur here.